packages/@aws-cdk/aws-ecs/lib/ec2/ec2-service.ts

```diff
@@ -128,9 +129,25 @@ export class Ec2Service extends BaseService implements IEc2Service {
   public static fromEc2ServiceArn(scope: Construct, id: string, ec2ServiceArn: string): IEc2Service {
     class Import extends Resource implements IEc2Service {
       public readonly serviceArn = ec2ServiceArn;
-      public readonly serviceName = Stack.of(scope).splitArn(ec2ServiceArn, ArnFormat.SLASH_RESOURCE_NAME).resourceName as string;
+      public readonly serviceName: string;
+      constructor() {
+        super(scope, id);
+        const resourceName = Stack.of(scope).splitArn(ec2ServiceArn, ArnFormat.SLASH_RESOURCE_NAME).resourceName as string;
+        if (Token.isUnresolved(ec2ServiceArn)) {
+          if (FeatureFlags.of(this).isEnabled(ECS_ARN_FORMAT_INCLUDES_CLUSTER_NAME)) {
+            const components = Fn.split(':', ec2ServiceArn);
+            const lastComponents = Fn.split('/', Fn.select(5, components));
+            this.serviceName = Fn.select(2, lastComponents);
+          } else {
+            this.serviceName = resourceName;
+          }
+        } else {
+          const resourceNameSplit = resourceName.split('/');
+          this.serviceName = resourceNameSplit.length === 1 ? resourceName : resourceNameSplit[1];
+        }
+      }
     }
-    return new Import(scope, id);
+    return new Import();
```

Comment:
Previously, passing the parameters was necessary because they were passed to the super class. The new implementation contains a constructor. In this case, the `super()` call has to be implemented in the constructor. Therefore, the parameters are not necessary anymore.

If you prefer the existing code (`return new Import(scope, id);`), I could implement it too. For example:

```ts
  constructor(importScope: Construct, importId: string) {
    super(importScope, importId);
    // ...
  }
}
return new Import(scope, id);
```

However, we have to use new attribute names. If I use `scope` instead of `importScope`, the following error is displayed:`'scope' is already declared in the upper scope.eslint@typescript-eslint/no-shadow`.

Which option do you prefer?

Reply:
No constructor is used anymore --> done.